src/Mod/Draft/DraftEdit.py

```diff
@@ -1136,23 +1143,35 @@ def updateCircle(self,v):
                     self.obj.FirstAngle = dangleF
                     self.obj.LastAngle = dangleL
                     FreeCAD.Console.PrintMessage("Press I to invert the circle\n")
-                    self.updateCirclePts()
+                    self.updateCircleTrackers(obj)
+
+    def getArcStart(self, obj):#Returns object midpoint
+        if Draft.getType(obj) == "Circle":
+            return self.pointOnCircle(obj, obj.FirstAngle)
+    
+    def getArcEnd(self, obj):#Returns object midpoint
+        if Draft.getType(obj) == "Circle":
+            return self.pointOnCircle(obj, obj.LastAngle)
 
     def getArcMid(self, obj):#Returns object midpoint
         if Draft.getType(obj) == "Circle":
             if obj.LastAngle>obj.FirstAngle:
-                midAngle=math.radians(obj.FirstAngle+(obj.LastAngle-obj.FirstAngle)/2)
+                midAngle=obj.FirstAngle+(obj.LastAngle-obj.FirstAngle)/2
```

Comment:
Divisions should use a floating point number, to make sure the result is a float. This is important in Python 2. In Python 3 is not needed but it doesn't hurt. Also add spaces between the operators for readability.
```py
midAngle = obj.FirstAngle + (obj.LastAngle - obj.FirstAngle)/2.0
```